### DISCUSSION TITLE
Flux Dev Fill Inpaint GGUF (Model + CLIP) Flow

### DISCUSSION THREAD

**Discussion by Hujikuio** [diStyApps/ComfyUI-disty-Flow#36] [General]
[tpxxa-flux-dev-fill-gguf-model-clip-inpainting.zip](https://github.com/user-attachments/files/18137813/tpxxa-flux-dev-fill-gguf-model-clip-inpainting.zip)
Same as preset, just replaced Double CLIP Loader node with the GGUF version.
Unzip, place in custom_nodes\ComfyUI-disty-Flow\web\flows

**Comment by Hujikuio** [comment]
I will try to upload more flows as I make them, this plugin is amazing so I definitely will eventually; I lose motivation quickly so no promises on timeline.

**Reply by diStyApps** [reply]
No worries do it at your own pace and enjoy it, i know the feeling very well. Flow is some kind of challenge to myself to stick to one project for a while.
Anyway you will be credited. I will add author field to the flows.
I plan letting people upload flows, but i need some quality control.
Also the linker needs to improve.

**Reply by Hujikuio** [reply]
+1 on QC, and no worries about crediting, I am for the people. Do what you want with anything I release and change it or make your own etc. Some people feel the need to credit others out of principle and if that's the case go for it. Some tie themselves up in knots because they're not sure at what point of Theseus' Ship is it not his anymore, so if that's you don't feel pressured about it, I feel weird about putting my name on anything that isn't from me 100% anyways, I'm just a silly end user making (probably inefficient) things ¯\\_(ツ)_/¯

**Comment by Hujikuio** [comment]
It is labeled v1 as I plan to replace the mask blur with gaussian blur (I find it better for inpainting, but I haven't played around with flux fill much yet, let me know if it's a bad idea.), and add crop and stitch in the pipeline so you can inpaint 4k images and stuff without melting your gpu (once again, if possible.)

**Reply by diStyApps** [reply]
Thank you very much, i missed i just updated all the GGUF flows too.
Am, its build that way for the live preview in painting, so i have to so some composting behind the back, only preview will not work properly. i will also look at crop at stitch maybe add support for that.

**Reply by Hujikuio** [reply]
I have [crop and stitch](https://github.com/lquesada/ComfyUI-Inpaint-CropAndStitch) working, but I haven't fully tested every feature so 
maybe there's something that would require additional eyes on, but it seems to be pretty plug and play to me. 

Overall though, it's a cool little suite with only four nodes. I'm very OCD about keeping my installs as efficient and "light" as possible, in regards to dependencies, node amount, and overlapping features, so it gets a pass in all those areas. It does what the image composite masked node does that you're already using, so the preset flows are find on that end, but I find the additional features packed into one install with only 4 nodes really makes the efficient and neat part of brain happy. It can outpaint, upscale/downscale, increase context/padding of masks, etc. 

I only use it in my upcoming flow because it does seem to be pretty plug and play and overall worth it. I try to keep in mind the custom nodes the presets already use and try to use those, and if I have to use other ones, I try to make sure it doesn't bloat a user's comfy install or have problematic dependencies such as xformers, onnx, etc. that are very sensitive on versions of cuda/pytorch/python aligning or breaking many commonly used dependencies. I run into stuff like that a lot with shared workflows, and it annoys me to no end. I either will release the others with huge disclaimers, or not at all.
***
While I'm on the topic of user experience (I can open a different discussion or request but I always forget or stop caring and it's best to write it out while it's fresh): I struggle with deciding if I want to make the workflow for me, with all the options I personally use, but then feel like I should include them all. Sometimes with long workflows it can get cluttered very easily regardless of whether I plan on others using them or not.

My ideal vision would be to make one workflow. It would have a "**Simple**" tab/section/whatever that the average user can enjoy, with the majority of "good enough" or "works in 95% of use cases" pre configured behind the scenes, preventing them from being overwhelmed. If a more advanced used that wants to take full advantage of the workflow, either because they are well-versed in it, or don't mind experimenting, they can switch to a "**Advanced**" tab where you can access the more under the hood and fine-tuning settings. Both "tabs" could have "**Optional**" parameters be able to be tucked away based on what the user is doing at that specific time, like if I have a workflow that can inpaint and outpaint, but they only plan on doing the former, there is no need to scroll past or see the outpaint parameters. 

Yes I am aware of the parallels that the above high-level example has between A1111 and ComfyUI, and it's why I love this node, it's already very much a "bridge" between the two worlds, and while Swarm is also another "bridge", I feel this project understands the average user a bit more. I mean your canvas implementation is already second only to Invoke, which I feel is an unfair comparison. Comfy at the time of writing has just come out with their improved beta canvas and while it is better than the old one (which is not saying much), yours is still leagues ahead.

I know a lot of these ideas and more are probably already floating around in your head, and that you already have a vision of where you want this project to go. I'm just offering up my own floating ideas/insights in case there's any ideas or suggestions that you haven't thought of, and I understand that you might already have 100 perfect ideas, but they all take time and we are single threaded machines at the end of the day. I'll selfishly and half-jokingly add that the only thing I really am looking forward to and hope comes out quicker is increased mobile support.

**Reply by Hujikuio** [reply]
Apologies for length, feel free to skim, I'm tired just looking at it.

**Reply by diStyApps** [reply]
Thank you very much and this is great read.
I also thought to make "Flows" more centralized so every workflow, you can have SD1.5/SDSXL/FLUX versions if it is possible and adding buttons/tabs/send to inpaint and such slimier to a1111.
Adding tabs for basic/advance is cool idea, i will add some sort of this functionality, we can discuss on this further, show me the workflows basic/simple/advance/more.

Also there possibility to chain workflows so a "Flow" can have multi workflows that adds different functionality like dedicated upscaler. 
The "Linker" should be refactored and among all other things i plan i really want to rewrite the linker a lot can be simplified more live approach.

But i am still focusing on the canvas to work better and adding outpainting better performance and other.
Flow was created mainly for better canvas functionality and experience,  it still not there but getting closer to my vision, i use the tools i make.
On every aspect of Flow i have 100 ideas, i wish i could implement them that fast.
The more ideas and other perspectives are great because it takes me out of my bubble, i do research of course it is not healthy to try always invent the wheel, i have over 20 experience graphic design and development, but still.
Brain storming is always welcome i find value in every discussion.

**Reply by Hujikuio** [reply]
If only we had infinite bodies with infinite time, I agree, always more to do and more that can be done. Your canvas is spectacular, can't wait to see how it evolves. The workflows are a work in progress, as I said before, I finish up and figure out a better / more efficient / more dynamic way to do things (It's why I cut myself off from comfy for a bit, the modularity was making me not actually generate anything lol). I'm going to "just do it" hopefully tonight or tomorrow. I want this to become on the level of the manager, rgthree, custom-scripts, etc.

The "newest" tech ones I have more or less ready to go are the canny and depth Flux workflows (both official model and official lora), but I struggle to see the point of the canny one as I don't think you can dynamically see the high/low threshold so you'd be kind of guessing, so I might put those on standby.

One more thing that is personally holding ME back from using certain models in normal txt2img mode, is no tag support. I'm not sure how easy that would be to implement from a .csv/.txt file, but just throwing it out there. Stuff like NoobAI's vPred models are EXTREMELY picky about tags so I can't really use it in Flow as one misplaced character or wrong tag will throw the whole image off.
